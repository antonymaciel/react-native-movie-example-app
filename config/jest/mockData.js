export const movie = {
    id: 0,
    year: '2019',
    imageUrl: 'https://image.tmdb.org/t/p/w500/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
    title: 'Spiderman',
    vote_average: '9.1',
    urlbackdropImage: 'https://image.tmdb.org/t/p/w500/kqjL17yufvn9OVLyXYpvtyrFfak.jpg', 
    generes: 'Aventure, Drama, Terror',  
    vote_count: '1200',
    overview: 'Amazing movie',
    backdrop_path: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
    poster_path: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
    genres: ['Adventure', 'Terror', 'Drama'],
    release_date: '2016-5-14'
}

export const review = {
    id: 0,
    author: 'Antony Maciel',
    url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
    content: 'Great Movie!, always has amazing content \n I love this movie'
}

export const movies = [
    {
        id: 1,
        poster_path: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
        title: 'XMan',
        vote_average: 7.8,
        release_date: '2016-5-14'
    },
    {     
        id: 2,
        year: '2019',
        imageUrl: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
        title: 'Superman',
        vote_average: 0.9,
        release_date: '2015-5-9'
    },
    {
        id: 3,
        year: '2019',
        imageUrl: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
        title: 'Spiderman',
        vote_average: 8.0,
        release_date: '2005-5-27'
    }
]


export const reviews = [
    {
        id: 1,
        author: 'Antony Maciel',
        url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
        content: 'Great Movie!!!'
    },
    {
        id: 2,
        author: 'Antony Maciel',
        url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
        content: 'Great Movie!, always has amazing content \n I love this movie'
    },
    {
        id: 3,
        author: 'Antony Maciel',
        url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
        content: '&^%$#@!'
    }
]

