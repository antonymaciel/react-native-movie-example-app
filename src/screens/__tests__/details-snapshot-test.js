import React from 'react';
import renderer from 'react-test-renderer';
import { movie } from '../../../config/jest/mockData';
import Details from '../details';



it('renders details screen using Snapshots', () => {
  expect(renderer.create(
    <Details
        urlbackdropImage={movie.urlbackdropImage}
        urlposterImage={movie.imageUrl}
        generes={movie.generes}
        title={movie.title}
        vote_count={movie.vote_count}
        vote_average={movie.vote_average}
        overview={movie.overview}
        goToReviews={jest.fn()}
    />
  )).toMatchSnapshot();
});

