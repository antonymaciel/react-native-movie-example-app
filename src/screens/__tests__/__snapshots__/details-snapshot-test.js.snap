// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders details screen using Snapshots 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View>
    <Image
      source={
        Object {
          "uri": "https://image.tmdb.org/t/p/w500/kqjL17yufvn9OVLyXYpvtyrFfak.jpg",
        }
      }
      style={
        Object {
          "backgroundColor": "black",
          "height": 533.6,
          "width": 750,
        }
      }
    />
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": "https://image.tmdb.org/t/p/w500/kqjL17yufvn9OVLyXYpvtyrFfak.jpg",
        }
      }
      style={
        Object {
          "backgroundColor": "black",
          "bottom": -177.86666666666667,
          "height": 355.73333333333335,
          "marginLeft": 20,
          "position": "absolute",
          "width": 237.15555555555557,
        }
      }
    />
    <Text
      style={
        Object {
          "bottom": 0,
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
          "left": 277.15555555555557,
          "marginBottom": 23,
          "position": "absolute",
          "right": 20,
          "textTransform": "uppercase",
        }
      }
    >
      Spiderman
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "left": 277.15555555555557,
        "marginTop": 23,
        "maxHeight": 154.86666666666667,
        "width": 462.84444444444443,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      1200
       
      <Text
        style={
          Object {
            "color": "black",
            "fontWeight": "300",
          }
        }
      >
        People watching
      </Text>
    </Text>
    <Text
      numberOfLines={3}
      style={
        Object {
          "color": "black",
          "fontWeight": "300",
        }
      }
    >
      Aventure, Drama, Terror
    </Text>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Text>
        9.1
      </Text>
      <View>
        <Text>
          9.1
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "margin": 23,
      }
    }
  >
    <Text
      numberOfLines={5}
      style={
        Object {
          "color": "#696969",
          "fontSize": 18,
          "fontWeight": "500",
        }
      }
    >
      Amazing movie
    </Text>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Show Reviews
        </Text>
      </View>
    </View>
  </View>
</View>
`;
